components/camel-univocity-parsers/src/main/java/org/apache/camel/dataformat/univocity/Marshaller.java

```diff
@@ -46,7 +46,9 @@
      */
     Marshaller(String[] headers, boolean adaptHeaders) {
         if (headers != null) {
-            this.headers.addAll(Arrays.asList(headers));
+            synchronized (this.headers) {

```

Comment:
I don't think this is necessary; I think it would make sense to treat headers as Immutable. So something like:

    this.headers = Collections.unmodifiableList(Arrays.asList(headers));